ansible_base/authentication/middleware.py

```diff
@@ -36,3 +42,16 @@ def process_request(self, request):
         if backend := request.session.get(BACKEND_SESSION_KEY, None):
             if backend in self.plugins:
                 request.session[BACKEND_SESSION_KEY] = "ansible_base.authentication.backend.AnsibleBaseAuth"
+
+
+class SocialExceptionHandlerMiddleware(SocialAuthExceptionMiddleware):
+    def get_redirect_uri(self, request, exception):
+        strategy = getattr(request, "social_strategy", None)
+        error_url = strategy.setting("LOGIN_ERROR_URL")
+        backend = getattr(request, "backend", None)
+        backend_name = getattr(backend, "name", "unknown-backend")
+        logger.error(f"Auth failure for backend {backend_name} - {repr(exception)}")
+        # UI expects auth_failed flag
+        url = urljoin(error_url, "/?auth_failed")
```

Comment:
I would suggest to use urlparse and add the param. This prevents all apps to specify their URL with params. 
Or, the `/?auth_failed`  could be part of dynamic settings and is possible to redefine it (voting for this one).

Reply:
Oh yeah that makes sense.  No problem, I'll just add the whole thing in the settings.